src/data_sources/INA226.cpp

```diff
@@ -0,0 +1,110 @@
+#include "INA226.h"
+
+#include "Registry.h"
+
+#include <map>
+
+std::map<std::string, std::string> counterNameToFileName;
+
+#if defined(__linux__)
+
+#include <fstream>
+#include <dirent.h>
+
+struct INA226Detail
+{
+	std::ifstream ifstrm;
+};
+
+std::string searchPath = "/sys/class/hwmon/";
+
+std::vector<std::string> INA226::detectAvailableCounters()
+{
+	std::vector<std::string> counters;
+
+	std::shared_ptr<DIR> dir(opendir(searchPath.c_str()), [](DIR *dir) {if (dir) closedir(dir);});
+	if (!dir) return counters;
+
+	struct dirent *entry;
+	while ((entry = readdir(dir.get()))) {
+		if (entry->d_type != DT_LNK) continue;
+
+		auto devicePath = searchPath + entry->d_name + "/";
+		std::ifstream nameFile(devicePath + "name");
+
+		std::string deviceName;
+		nameFile >> deviceName;
+
+		if (deviceName == "ina226") {
+			std::string counterName = "power-";
+			counterName += entry->d_name;
+
+			counters.push_back(counterName);
+			counterNameToFileName.insert({counterName, devicePath + "power1_input"});
+		}
+	}
+
+	return counters;
+}
+
+PowerSample INA226::read() {
+	m_detail->ifstrm.seekg(std::ios_base::beg);
+	int val;
+	m_detail->ifstrm >> val;
+
+	return PowerSample(units::power::microwatt_t(val));
```

Comment:
Is the microwatt resolution actually supported by the chip, our just how the kernel driver exposes it?
If yes, you can alternatively implement `read_mW_string`, so in case of continuous printing we don't convert back and forth between string and double.

Reply:
It looks like the chip supports microwatt resolution. For instance, on Board96 (with PYNQ Linux) the power LSB is 6250µW, which also happens to be the default value for the Linux INA266 driver. I'd just let it stay like this for now.